libraries/chain_kv/include/b1/chain_kv/chain_kv.hpp

```diff
@@ -302,28 +305,46 @@ struct database {
    std::unique_ptr<rocksdb::DB> rdb;
 
    database(const char* db_path, bool create_if_missing, std::optional<uint32_t> threads = {},
-            std::optional<int> max_open_files = {}) {
-
-      rocksdb::Options options;
-      options.create_if_missing                    = create_if_missing;
-      options.level_compaction_dynamic_level_bytes = true;
-      options.bytes_per_sync                       = 1048576;
+            std::optional<int> max_open_files = {}, std::optional<boost::filesystem::path> options_file_name = {}) {
+      rocksdb::DB* p;
 
-      if (threads)
-         options.IncreaseParallelism(*threads);
+      if (options_file_name) {
+         // load the options file
+         rocksdb::DBOptions loaded_db_opt;
+         std::vector<rocksdb::ColumnFamilyDescriptor> loaded_cf_descs;
+         check(LoadOptionsFromFile(options_file_name->string(), rocksdb::Env::Default(), &loaded_db_opt, &loaded_cf_descs), "database::database: rocksdb::LoadLatestOptions: ");
+         loaded_db_opt.env->SetBackgroundThreads(loaded_db_opt.max_background_jobs, rocksdb::Env::LOW); // low priority background threads pool
+         loaded_db_opt.env->SetBackgroundThreads(1, rocksdb::Env::HIGH); // high priority background threads pool
+
+         // open rocksdb using the loaded options
+         std::vector<rocksdb::ColumnFamilyHandle*> handles;
+         check(rocksdb::DB::Open(loaded_db_opt, db_path, loaded_cf_descs, &handles, &p), "database::database:rocksdb::DB::Open (options files): ");
+      } else {
+         rocksdb::Options options;
+         options.create_if_missing = create_if_missing;
 
-      options.OptimizeLevelStyleCompaction(256ull << 20);
+         if (threads) {
+            options.IncreaseParallelism(*threads);
+         } else {
+            options.IncreaseParallelism(20);
+         }
+         if (max_open_files) {
+            options.max_open_files = *max_open_files;
+         } else {
+            options.max_open_files = 768;
+         }
 
-      if (max_open_files)
-         options.max_open_files = *max_open_files;
+         options.compaction_style = rocksdb::kCompactionStyleLevel;
+         options.level0_file_num_compaction_trigger = 10;
+         options.level0_slowdown_writes_trigger = 20;
+         options.level0_stop_writes_trigger = 40;
+         options.write_buffer_size = 256 * 1024 * 1024;
+         options.target_file_size_base = 256 * 1024 * 1024;
+         options.max_bytes_for_level_base = 256 * 1024 * 1024;  // L1 max size
```

Comment:
I am curious about how you come up with these numbers?

Reply:
They were from RocksDB Performance Tuning Guide for a typical setting. I will add comments explaining them.